composer/trainer/trainer.py

```diff
@@ -1222,6 +1222,10 @@ def __init__(
         warnings.filterwarnings(action='ignore', message='torch.cuda.amp.GradScaler')
         self.state.scaler = ClosureGradScaler() if self._use_closures() else GradScaler()
 
+        if self.fsdp_config is not None:
+            from torch.distributed.fsdp.sharded_grad_scaler import ShardedGradScaler
+            self.state.scaler = ShardedGradScaler()
+
```

Comment:
I believe the FSDP config needs to be gated. 

also can we confirm closures aren't used here (just raise ValueError if they are, though we should never reach this state)